### PR TITLE
Lookup lemmatizer

### PR DESCRIPTION
**Pull request by qeterme**
Added Lookup Lemmatizer and also its usage in the `hu_core_news_lg` model, it returns a lemma from the token and its POS tag. 
I also added the lemma smoother to the `hu_core_news_lg` model and now it has an accuracy of 97.36%.

### CODE REVIEW

File: hu_core_news_lg/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## `hu_core_news_lg`
 
+## 3.3.1
+### Added
+- Added lookup lemmatizer before edit tree lemmatizer in the pipeline
+- Added lemma smoother after edit tree lemmatizer in the pipeline
+
+## 3.3.0
```
Comment (oroszgy) [inline hu_core_news_lg/CHANGELOG.md:15]: Thanks for fixing this for me!

File: huspacy/tools/components.py [hunk 2]
```diff
@@ -69,6 +70,61 @@ def initialize(
...
+        self._lookups: Optional[Lookups] = lookups
+        self.scorer = scorer
+        self.source = source
+
+    def __call__(self, doc: Doc) -> Doc:
+        assert self._lookups is not None, "Lookup table should be initialized first"
+
+        token: Token
+        for token in doc:
+            if token.text in self._lookups.get_table(f"lemma_lookups_{token.tag_}"):
```
Comment (oroszgy) [inline huspacy/tools/components.py:98]: Just a minor thing: this is very readable, but not very memory efficient. I might consider using a shorter table name. 

Is there a naming convention in spacy for that?

File: scripts/train_lookup_lemmatizer.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+from tqdm import tqdm
+from conllu import parse_incr
+
+from spacy.lookups import Lookups, Table
+
+app = typer.Typer()
+
+
+@app.command()
+def main(input_file: str, output_path: Path, min_occurrences: int = 1):
```
Comment (oroszgy) [inline scripts/train_lookup_lemmatizer.py:14]: 1 seems to be a reasonable choice, but I would be curious how does the model perform with values like 3,5,10. (My guess is that much worse.)

File: scripts/train_lookup_lemmatizer.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    token_lemma_occurrences = dict()
+
+    with open(input_file) as inp:
+        for sentence in tqdm(list(parse_incr(inp))):
+            for token in sentence:
+                if token["upos"] not in token_lemma_occurrences.keys():
```
Comment (oroszgy) [inline scripts/train_lookup_lemmatizer.py:20]: This logic can be simplified with `collections.Defaultdict` or `collections.Counter`. I'd suggest to use the latter one.

**Review by oroszgy — CHANGES_REQUESTED** [review]
Nice job again! 

Could you please add these lemma smoothing steps to the `trf` model as well?

File: huspacy/tools/components.py (hunk 2 above)
Comment (qeterme) [reply huspacy/tools/components.py:98]: In the Polish 3.2.x model they're using this naming convention. In theory, this should allow us to use the official Lemmatizer pipe, but I cannot see how.

**Comment by qeterme** [comment]
> Could you please add these lemma smoothing steps to the `trf` model as well?
Yes, I've added to it, and now it has an accuracy of 98.75%

File: huspacy/tools/components.py (hunk 2 above)
Comment (oroszgy) [reply huspacy/tools/components.py:98]: Thanks for the info. Than let's stick w/ the naming conventions.

File: scripts/train_lookup_lemmatizer.py (hunk 4 above)
Comment (qeterme) [reply scripts/train_lookup_lemmatizer.py]: Replaced!

File: scripts/train_lookup_lemmatizer.py (hunk 3 above)
Comment (qeterme) [reply scripts/train_lookup_lemmatizer.py:15]: Yes, it got worse. I tested on the `hu_core_news_lg` model with the full pipeline. 
Here are the results:
1: 97.36%
3: 96.97%
5: 96.94%
10: 96.95%

**Review by oroszgy — APPROVED** [review]

**Pull request merged**